core/commands/repo.go

```diff
@@ -290,9 +287,12 @@ var repoVerifyCmd = &cmds.Command{
 				out <- &VerifyProgress{Progress: i}
 			}
 			if fails == 0 {
+				log.Debug("Run.go.else: all ok")
 				out <- &VerifyProgress{Message: "verify complete, all blocks validated."}
 			} else {
-				out <- &VerifyProgress{Message: "verify complete, some blocks were corrupt."}
+				//out <- &VerifyProgress{Message: "verify complete, some blocks were corrupt."}
+				log.Debugf("Run.go.else: %T.SetError", res)
+				res.SetError(fmt.Errorf("verify complete, some blocks were corrupt"), cmdsutil.ErrNormal)
```

Comment:
I'm fine with not changing the old stuff, but theres no need to make it variadic. Just change the signature to:
```go
func (res Response) SetError(msg interface{}, code ErrorType)
```
Then in the function you would type switch msg to either an error or a string, and just `fmt.Sprint` anything else that wasnt one of those.

But, like it said, don't bother spending time changing the legacy stuff :)